### PR TITLE
Oauth bearer sig

### LINKED ISSUE #97: Check bearer token JWT headers according to spec
This means that the x5c header must also be validated and that the leaf in the x5c field is a known vendor!

- [x] §5.2.1.1
- [x] §5.2.1.2 -> #115 
- [x] §5.2.1.3
- [x] §5.2.1.4
- [x] §5.2.1.5
- [x] §5.2.1.6 -> #125 
- [x] §5.2.1.7 -> #126  
- [x] §5.2.1.8

### PR DESCRIPTION
**Pull request by woutslakhorst**
Closes #97

2 validations require additional functionality from the registry:
- oauth scope <> consent scope <> services scope relations
- checking jwt.aud with services and endpoints definition

Issues have been created to fix missing validations: #115, #125, #126
This is not blocking since current rollout features trusted parties.

### CODE REVIEW

**Comment by woutslakhorst** [comment]
build failure due to some mac <> linux difference

File: pkg/services/oauth/oauth.go [hunk 1]
```diff
@@ -56,18 +85,39 @@ func (s *service) Configure() (err error) {
...
+	}
+
+	// check the actor against the registry, according to RFC003 §5.2.1.3
+	if err = s.validateIssuer(jwtBearerToken); err != nil {
+		return nil, err
+	}
+
+	// check the maximum validity, according to RFC003 §5.2.1.4
+	if jwtBearerToken.ExpiresAt-jwtBearerToken.IssuedAt > OauthBearerTokenMaxValidity {
+		return nil, errors.New("JWT validity to long")
```
Comment (reinkrul) [inline pkg/services/oauth/oauth.go:117]: to -> too

File: pkg/services/oauth/oauth.go [hunk 2]
```diff
@@ -84,6 +144,78 @@ func (s *service) CreateAccessToken(request services.CreateAccessTokenRequest) (
 	return &services.AccessTokenResult{AccessToken: accessToken}, nil
 }
 
+// check the actor against the registry, according to RFC003 §5.2.1.3
+// we do this by getting the validation chain for the certificate in the x5c header and check the vendorID SAN from the root
+// with the vendorId of the actor
+func (s *service) validateIssuer(jwtBearerToken *services.NutsJwtBearerToken) error {
+	validationTime := time.Unix(jwtBearerToken.IssuedAt, 0)
+
+	actorPartyID, err := core.ParsePartyID(jwtBearerToken.Issuer)
```
Comment (reinkrul) [inline pkg/services/oauth/oauth.go:153]: do we have enough logging (e.g. of issuer PartyIDs which proved to be invalid) to analyse production issues? Or is the issuer mentioned in the returned error?

File: pkg/services/oauth/oauth.go [hunk 3]
```diff
@@ -84,6 +144,78 @@ func (s *service) CreateAccessToken(request services.CreateAccessTokenRequest) (
...
+	chains, err := s.crypto.TrustStore().VerifiedChain(jwtBearerToken.SigningCertificate, validationTime)
+	if err != nil || len(chains) == 0 {
+		return fmt.Errorf(errInvalidIssuerFmt, err)
+	}
+	match := false
+	for _, chain := range chains {
+		root := chain[len(chain)-1]
+		v, err := cert.VendorIDFromCertificate(root)
+		if err != nil {
+			fmt.Errorf("no vendorID in SAN: %w", err)
```
Comment (reinkrul) [inline pkg/services/oauth/oauth.go:170]: return

**Review by reinkrul — CHANGES_REQUESTED** [review]

**Comment by woutslakhorst** [comment]
Covered by #131

**Pull request closed without merging**